staging/src/k8s.io/apiserver/pkg/server/routes/index.go

```diff
@@ -65,5 +69,18 @@ type IndexLister struct {
 
 // ServeHTTP serves the available paths.
 func (i IndexLister) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	responsewriters.WriteRawJSON(i.StatusCode, metav1.RootPaths{Paths: i.PathProvider.ListedPaths()}, w)
+	userTenant, _ := apifilters.GetTenantFromQueryThenContext(r)
+	tenantedPathMark := "?tenant="
+	tenantPathSuffix := tenantedPathMark + userTenant
+	rootPaths := []string{}
+	for _, path := range i.PathProvider.ListedPaths() {
+		if !strings.Contains(path, tenantedPathMark) {
+			rootPaths = append(rootPaths, path)
+		} else {
+			if strings.HasSuffix(path, tenantPathSuffix) {
+				rootPaths = append(rootPaths, strings.TrimSuffix(path, tenantPathSuffix))
```

Comment:
add a unit test for this ServeHTTP if there isn't one already

Reply:
Added.